src/pkg/cli/client/byoc/gcp/byoc.go

```diff
@@ -647,16 +675,162 @@ func (b *ByocGcp) PutConfig(ctx context.Context, req *defangv1.PutConfigRequest)
+
+	newQueryFragment = CreateServiceStatusRequestUpdate(req.Project, req.Etag, req.Services) // CloudBuild logs
+	query = ConcatQuery(query, newQueryFragment)
+
+	newQueryFragment = CreateServiceStatusReponseUpdate(req.Project, req.Etag, req.Services) // CloudBuild logs
+	query = ConcatQuery(query, newQueryFragment)
+
+	return query
+}
+
+func LogEntryToString(logEntry *loggingpb.LogEntry) (string, string, error) {
+	result := ""
+	emptySpace := strings.Repeat(" ", len(time.RFC3339)+1) // length of what a time stampe would be
+	logTimestamp := emptySpace
+	if logEntry.Timestamp != nil {
+		logTimestamp = logEntry.Timestamp.AsTime().Local().Format(time.RFC3339) + " "
+	}
+
+	switch {
+	case logEntry.GetJsonPayload() != nil:
+		result += logTimestamp + logEntry.GetJsonPayload().String()
+	case logEntry.GetProtoPayload() != nil:
+		auditLog := &auditpb.AuditLog{}
+		if err := logEntry.GetProtoPayload().UnmarshalTo(auditLog); err != nil {
+			return logTimestamp, "", err
+		}
+
+		// we do not know the reason for no status events we will log it here (update in future)
+		if auditLog.Status == nil {
+			return logTimestamp, "<No Status>", nil
+		}
+		return logTimestamp, auditLog.Status.Message, nil
+	case logEntry.GetTextPayload() != "":
+		return logTimestamp, logEntry.GetTextPayload(), nil
+	}
+
+	return logTimestamp, result, nil
+}
+
+func LogEntriesToString(logEntries []*loggingpb.LogEntry) string {
+	result := ""
+	for _, logEntry := range logEntries {
+		logTimestamp, msg, err := LogEntryToString(logEntry)
+		if err != nil {
+			continue
+		}
+		result += logTimestamp + " " + msg + "\n"
+	}
+	return result
+}
+
+func (b *ByocGcp) query(ctx context.Context, client *logging.Client, projectId string, query string) ([]*loggingpb.LogEntry, error) {
+	var entries []*loggingpb.LogEntry
+
+	req := &loggingpb.ListLogEntriesRequest{
+		ResourceNames: []string{"projects/" + projectId}, // Required
+		Filter:        query,                             // Query string
+		OrderBy:       "timestamp desc",                  // Optional: Sort by timestamp
+		PageSize:      50,                                // Number of entries per page
+	}
+
+	term.Debugf("Querying logs with filter: \n %s", query)
+	resp := client.ListLogEntries(ctx, req)
+	defer client.Close()
+
+	for {
+		// aggregate log entries
+		for {
+			entry, err := resp.Next()
+			if err != nil {
+				if err == iterator.Done {
+					break
+				}
+				return nil, err
+			}
+
+			entries = append(entries, entry)
+		}
+
+		// set request for next page
+		if resp.PageInfo().Token == "" {
+			break
+		}
+		req.PageToken = resp.PageInfo().Token
+	}
+
+	return entries, nil
+}
+
+func (b *ByocGcp) PopulateWithCdJobInfo(etag string) (string, *timestamppb.Timestamp, error) {
+	execution, err := b.driver.FindExecutionWithEtag(etag)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not find job with etag %s: %v", etag, annotateGcpError(err))
+	}
+	return execution.Name, execution.CreateTime, nil
```

Comment:
Do we need this function? I found the name to be confusing when looking at a call site, and now looking at the definition, I don't think it's adding any value.